packages/showcase/examples/responsive-vis/responsive-vis-example.js

```diff
@@ -18,117 +18,108 @@
+    ? getBarFeatures
+    : getScatterplotFeatures)(featuresProps);
 
-    return (
-      <div className="responsive-controls">
-        <div className="points-per-pixel-label">{`Points Per Pixel: ${ppp}`}</div>
-        <div className="features-label">
-          {`Features: ${Object.keys(featuresToRender).join(', ')}`}
-        </div>
-        <div className="chart-type-selector">
-          <div
-            className={
-              chartType === 'scatterplot'
-                ? 'selected-chart-type'
-                : 'unselected-chart-type'
-            }
-            onClick={this.handleTypeClick('scatterplot')}
-          >
-            Scatterplot
+  return (
+    <div className="responsive-vis-example">
+      <div className="controls-wrapper">
+        <div className="responsive-controls">
+          <div className="points-per-pixel-label">{`Points Per Pixel: ${ppp}`}</div>
+          <div className="features-label">
+            {`Features: ${Object.keys(featuresToRender).join(', ')}`}
           </div>
-          <div
-            className={
-              chartType === 'barChart'
-                ? 'selected-chart-type'
-                : 'unselected-chart-type'
-            }
-            onClick={this.handleTypeClick('barChart')}
-          >
-            BarChart
+          <div className="chart-type-selector">
+            <div
+              className={
+                chartType === 'scatterplot'
+                  ? 'selected-chart-type'
+                  : 'unselected-chart-type'
+              }
+              onClick={handleTypeClick('scatterplot')}
+            >
+              Scatterplot
+            </div>
+            <div
+              className={
+                chartType === 'barChart'
+                  ? 'selected-chart-type'
+                  : 'unselected-chart-type'
+              }
+              onClick={handleTypeClick('barChart')}
+            >
+              BarChart
+            </div>
           </div>
-        </div>
-        {`Data Size: ${~~Math.pow(10, dataSize)}`}
-        <input
-          onChange={e => {
-            this.setState({
-              dataSize: e.target.value,
-              data: createData(
-                ~~Math.pow(10, e.target.value),
-                this.state.chartType === 'barChart'
-              )
-            });
-          }}
-          type="range"
-          min={1}
-          max={6}
-          step={0.1}
-          value={dataSize}
-        />
-
-        {`Visualization size: ${visSize}`}
-        <input
-          onChange={e => this.setState({visSize: ~~e.target.value})}
-          type="range"
-          min={100}
-          max={1000}
-          value={visSize}
-        />
-      </div>
-    );
-  }
+          {`Data Size: ${~~Math.pow(10, dataSize)}`}
+          <input
+            onChange={e => {
+              setDataSize(e.target.value);
+              setData(
+                createData(
+                  ~~Math.pow(10, e.target.value),
```

Comment:
I've never seen `~~` being used before

Reply:
This example is derived from http://nrabinowitz.github.io/rdv/ and I was just trying to copy as much of his code as I could. I'm not a huge fan of ascii art operators like that, but felt appropriate in keeping with the style of the copied code. shurggie

Reply:
I'm fine with it... I just have never seen it before.
I had completely forgotten about the `~arr.indexOf()` to check if the value is included.
No need to change.